tiny_dnn/core/framework/tensor.h

```diff
@@ -392,21 +375,23 @@ class Tensor {
    * are bigger than the current dimensions number. Also raises an exception
    * when the requested view size is not feasible.
    */
-  Tensor subview_impl(std::initializer_list<size_t> const &start,
-                      std::initializer_list<size_t> const &new_shape) {
-    if (start.size() > kDimensions || new_shape.size() > kDimensions) {
+  Tensor subview_impl(const std::vector<size_t> &start,
+                      const std::vector<size_t> &new_shape) {
+    if (start.size() > dims() || new_shape.size() > dims()) {
       throw nn_error("Overpassed number of existing dimensions.");
     }
 
     // compute the new offset and check that it's feasible to create
     // the new view.
-    // TODO(edgarriba/randl): add proper tests to this
     const size_t new_offset = offset_ + compute_offset(start, shape_);
     if (new_offset + product(new_shape) > size_) {
       throw nn_error("Cannot create a view from this tensor");
     }
 
-    return Tensor(storage_ptr_, new_offset, new_shape);
+    // computes the stride
+    const size_t new_stride = std::abs(shape_[dims() - 1] - new_shape[1]);
```

Comment:
This is a very trivial thing.Since definition of `Tensor::dims` method is `size_t dims() const { return shape_.size(); }`,
This line can also be written in this way.
```cpp
const size_t new_stride = std::abs(shape_.back() - new_shape[1]);
```